src/main/java/services/ClusterService.java

```diff
@@ -291,6 +291,20 @@ default void inputDetail() {
       * </action>
       * ----
       *
+      * After starting the upgrade, use a request body like this to update the progress to 15%:
+      *
+      * [source,xml]
+      * ----
+      * <action>
+      *     <upgrade_action>
+      *         update
+      *     </upgrade_action>
+      *     <upgrade_progress>
+      *         15
+      *     </upgrade_progress>
+      * </action>
```

Comment:
Consider including correlation ID  as unique identifier for the upgrade  (i.e. `upgradeCorrelationId`).
This would create a persistent link between cluster and upgrade that can be used i.e.  for filtering in the UI. Additionally, it could be used as a token for  `update` or `finish` actions - they should be accepted only if they contain matching correlation ID.

Reply:
Added `upgradeCorrelationId` to the Cluster and a matching optional `correlationId` to the service.  If not specified explicitly, the REST service can pick it up from the http headers.  cluster_upgrade ansible role always includes the correlation id in rest the header.